CONTRIBUTING.md

```diff
@@ -1,22 +1,78 @@
+* A "major" feature or bug fix is defined as any change that is > 100 lines of code (not including tests) or changes user-facing behavior (e.g., breaking API changes). Please read [Proposing Major Changes to Hyperspace](#proposing-major-changes-to-hyperspace) before you begin any major work.	
+* Once you are ready, you can create a [PR](https://github.com/microsoft/hyperspace/pulls) and the committers will help reviewing your PR.	
 
+**Coding Style: ** Please review our [coding guidelines](/docs/coding-guidelines/scala-coding-style.md).	
 
-### Coding Style
+# Proposing Major Changes to Hyperspace
 
-Please review our [coding guidelines](/docs/coding-guidelines/scala-coding-style.md).
+## Introduction	
+
+The development process in Hyperspace is design-driven. If you intend of making any significant changes, please consider discussing with the Hyperspace community first (and sometimes formally documented), before you open a PR.
+
+The rest of this document describes the process for proposing, documenting and implementing changes to the Hyperspace project.
+
+To learn about the motivation behind Hyperspace, see the talk [Hyperspace: An Indexing Subsystem for Apache Spark](https://www.youtube.com/watch?v=ofn53mT7H6c) from Spark+AI Summit 2020.	
+
+## The Proposal Process	
+
+The process outlined below is for reviewing a proposal and reaching a decision about whether to accept/decline a proposal.	
+
+  1. The proposal author [creates a brief issue](https://github.com/microsoft/hyperspace/issues/new?assignees=&labels=untriaged%2C+proposal&template=design-template.md&title=%5BPROPOSAL%5D%3A+) describing the proposal.	
+     > **Note: There is no need for a design document at this point.	
+  2. A discussion on the issue will aim to triage the proposal into one of three outcomes:	
+     - Accept proposal	
+     - Decline proposal	
+     - Ask for a design doc	
+     If the proposal is accepted/declined, the process is done. Otherwise, the discussion is expected to identify concerns that should be addressed in a more detailed design document.	
+  3. The proposal author [writes a design doc](#writing-a-design-document) to work out details of the proposed design and address the concerns raised in the initial discussion.	
+  4. Once comments and revisions on the design document are complete, there is a final discussion on the issue to reach one of two outcomes:	
+     - Accept proposal	
+     - Decline proposal	
+     	
+After the proposal is accepted or declined (e.g., after Step 2 or Step 4), implementation work proceeds in the same way as any other contribution. 	
+
+> **Tip:** If you are an experienced committer and are certain that a design doc will be required for a particular proposal, you can skip Step 1 and 2 and just include the doc PR with the initial issue.	
```

Comment:
How would you do this if the design doc name template is `docs/design/GITHUB-ISSUE-NUMBER-shortname.md`?  In this case there would be no GitHub issue??

Reply:
Great catch! Fixed it.